pyomo/contrib/doe/measurements.py

```diff
@@ -25,328 +25,311 @@
+        self_define_res: a ``list`` of ``string``, containing the variable names with indexes,
+            for e.g. "C['CA', 23, 0]".
         """
-        if mode == 'simultaneous_finite':
-            # check extra index
-            if self.ind_string in j:
-                measure_name = j.split(self.ind_string)[0]
-                measure_index = j.split(self.ind_string)[1]
-                if type(self.name_and_index[measure_name][0]) is str:
-                    measure_index = '"' + measure_index + '"'
-                if t in self.flatten_measure_timeset[j]:
-                    up_C = (
-                        'm.'
-                        + measure_name
-                        + '['
-                        + str(scenario_all['jac-index'][p][0])
-                        + ','
-                        + measure_index
-                        + ','
-                        + str(t)
-                        + ']'
-                    )
-                    lo_C = (
-                        'm.'
-                        + measure_name
-                        + '['
-                        + str(scenario_all['jac-index'][p][1])
-                        + ','
-                        + measure_index
-                        + ','
-                        + str(t)
-                        + ']'
-                    )
-                else:
-                    legal_t = False
-            else:
-                up_C = (
-                    'm.'
-                    + j
-                    + '['
-                    + str(scenario_all['jac-index'][p][0])
-                    + ','
-                    + str(t)
-                    + ']'
-                )
-                lo_C = (
-                    'm.'
-                    + j
-                    + '['
-                    + str(scenario_all['jac-index'][p][1])
-                    + ','
-                    + str(t)
-                    + ']'
-                )
-
-            return up_C, lo_C, legal_t
-
-        elif mode == 'sequential_finite':
-            if self.ind_string in j:
-                measure_name = j.split(self.ind_string)[0]
-                measure_index = j.split(self.ind_string)[1]
-                if type(self.name_and_index[measure_name][0]) is str:
-                    measure_index = '"' + measure_index + '"'
-                if t in self.flatten_measure_timeset[j]:
-                    string_name = (
-                        'mod.'
-                        + measure_name
-                        + '[0,'
-                        + str((measure_index))
-                        + ','
-                        + str(t)
-                        + ']'
-                    )
-            else:
-                string_name = 'mod.' + j + '[0,' + str(t) + ']'
-
-            return string_name
-
-    def check_subset(self, subset, throw_error=True, valid_subset=True):
+        super().set_variable_name_list(self_define_res)
+        self.name = self.variable_names
```

Comment:
Why is this necessary? Why not just use self.variable_names throughout this class. We could then eliminate this method.